internal/commands/save.go

```diff
@@ -31,247 +24,105 @@ func (cmd *SaveCmd) Run(ctx context.Context, globals *Globals) error {
 
 	log.Info().Str("version", globals.Version).Msg("Running SaveCmd")
 
-	// Augment `cli.Caches` with template values.
-	caches, err := configuration.ExpandCacheConfiguration(globals.Caches)
-	if err != nil {
-		return fmt.Errorf("failed to load cache configuration: %w", err)
-	}
-
-	for _, cache := range caches {
-		if len(cmd.Ids) > 0 && !slices.Contains(cmd.Ids, cache.ID) {
-			log.Debug().Str("id", cache.ID).Msg("Skipping cache save for ID")
-			continue
+	// Get list of all caches
+	caches := globals.CacheClient.ListCaches()
+
+	// Filter by IDs if specified
+	var cacheIDs []string
+	if len(cmd.Ids) > 0 {
+		cacheIDs = cmd.Ids
+	} else {
+		// Save all caches
+		for _, c := range caches {
+			cacheIDs = append(cacheIDs, c.ID)
 		}
+	}
 
-		if err := cache.Validate(); err != nil {
-			return fmt.Errorf("cache validation failed for ID %s: %w", cache.ID, err)
-		}
+	// Save each cache
+	for _, cacheID := range cacheIDs {
+		span.SetAttributes(attribute.String("cache_id", cacheID))
 
-		if err := cmd.saveCache(ctx, cache, globals); err != nil {
+		if err := cmd.saveCache(ctx, cacheID, globals.CacheClient, globals); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
-func (cmd *SaveCmd) saveCache(ctx context.Context, cache cache.Cache, globals *Globals) error {
+func (cmd *SaveCmd) saveCache(ctx context.Context, cacheID string, cacheClient *zstash.Cache, globals *Globals) error {
 	ctx, span := trace.Start(ctx, "saveCache")
 	defer span.End()
```

Comment:
Should the spans here also be moved up into the library code out of the internal code?